server/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```diff
@@ -609,32 +609,35 @@ public static FieldSortBuilder getPrimaryFieldSortOrNull(SearchSourceBuilder sou
         MappedFieldType fieldType,
         FieldSortBuilder sortBuilder
     ) throws IOException {
-        String fieldName = fieldType.name();
-        if (PointValues.size(reader, fieldName) == 0) {
-            return null;
-        }
-        if (fieldType instanceof NumberFieldType) {
-            NumberFieldType numberFieldType = (NumberFieldType) fieldType;
-            Number minPoint = numberFieldType.parsePoint(PointValues.getMinPackedValue(reader, fieldName));
-            Number maxPoint = numberFieldType.parsePoint(PointValues.getMaxPackedValue(reader, fieldName));
-            switch (IndexSortConfig.getSortFieldType(sortField)) {
-                case LONG:
-                    return new MinAndMax<>(minPoint.longValue(), maxPoint.longValue());
-                case INT:
-                    return new MinAndMax<>(minPoint.intValue(), maxPoint.intValue());
-                case DOUBLE:
-                    return new MinAndMax<>(minPoint.doubleValue(), maxPoint.doubleValue());
-                case FLOAT:
-                    return new MinAndMax<>(minPoint.floatValue(), maxPoint.floatValue());
-                default:
-                    return null;
+        if (fieldType instanceof NumberFieldType || fieldType instanceof DateFieldType) {
```

Comment:
We don't need to introduce this new type check here.

Reply:
@dnhatn Thanks for the so fast review! I checked this because i find that any other fieldtype (e.g. ScaledFloatFieldType) should return null in this method, So minPackedValue does not need to computed for these types, but it's ok to remove this check because it seems not related to this PR. fixed :)